src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -39,7 +58,11 @@
      * @return map {user email : user}
      */
     public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+        Map<String, User> usersMap = new HashMap<>();
+        for(int i=0; i<users.size(); ++i){
```

Comment:
Same applies here, forget about loops when doing this task.
Find another way to group users by email.